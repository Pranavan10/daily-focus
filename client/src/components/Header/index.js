import React from "react";
import Button from "@material-ui/core/Button";
import { IconButton, makeStyles } from "@material-ui/core";
import PersonIcon from "@material-ui/icons/Person";
import styles from "./style.module.scss";
import FocusLogo from "../../images/focus-logo.svg";
import SearchBar from "../SearchBar";
import DateTime from "../DateTime";

const useStyles = makeStyles({
    root: {
        background: "#30A0F5",
        borderRadius: 50,
        padding: "4px 25px",
    },
    label: {
        textTransform: "capitalize",
        fontSize: "1.25rem",
    },
    icon: {
        "background": "#30A0F5",
        "padding": 10,
        "&:hover": {
            backgroundColor: "#303f9f",
        },
    },
    iconLabel: {
        fill: "white",
        fontSize: "2rem",
    },
});

export default function Header() {
    const classes = useStyles();

    return (
        <header className={styles.header}>
            <div className={styles.focusLogo}>
                <img style={{ width: 175 }} src={FocusLogo} alt="FOCUS" />
            </div>

            <div className={styles.dateTime}>
                <DateTime />
            </div>

            <div className={styles.searchBar}>
                <SearchBar />
            </div>

            <div className={styles.addWidget}>
                <Button
                    variant="contained"
                    color="primary"
                    classes={{ root: classes.root, label: classes.label }}
                >
                    Add Widget
                </Button>
            </div>

            <div className={styles.userIcon}>
                <IconButton aria-label="user menu" classes={{ root: classes.icon }}>
                    <PersonIcon classes={{ root: classes.iconLabel }} />
                </IconButton>
            </div>
        </header>
    );
}
