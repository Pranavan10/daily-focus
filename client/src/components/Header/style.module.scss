.header {
    display: flex;
    height: 100px;
    justify-content: space-around;
    align-items: center;
    font-size: 1.5rem;
    background-color: white;
    -webkit-box-shadow: 0px 7px 20px 0px rgba(0, 0, 0, 0.1);
    -moz-box-shadow: 0px 7px 20px 0px rgba(0, 0, 0, 0.1);
    box-shadow: 0px 7px 20px 0px rgba(0, 0, 0, 0.1);

    .focusLogo {
        flex: 1 1 0;
        text-align: center;
    }

    .dateTime {
        flex: 1 1 0;
    }
    .searchBar {
        flex: 1.5 1 0;
    }

    .addWidget {
        flex: 0.75 1 0;
        text-align: center;
    }

    .userIcon {
        flex: 0.4 2 0;
    }
}
