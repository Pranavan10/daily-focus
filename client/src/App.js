import React from "react";
import { BrowserRouter, Redirect, Route, Switch } from "react-router-dom";

import LandingPage from "./components/LandingPage";
import Header from "./components/Header";
import "./App.css";

function App() {
    return (
        <BrowserRouter>
            <Switch>
                <Route exact path="/">
                    <Redirect to="/signup" />
                </Route>

                <Route exact path="/signup">
                    <LandingPage />
                </Route>

                <Route exact path="/home">
                    {/* Replace following component with actual Home component when created */}
                    <Header />
                </Route>
            </Switch>
        </BrowserRouter>
    );
}

export default App;
