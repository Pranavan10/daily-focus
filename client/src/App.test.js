import React from "react";
import { shallow } from "enzyme";
import ShallowRenderer from "react-test-renderer/shallow";

import App from "./App";
import Header from "./components/Header";

let component;

beforeEach(() => {
    component = shallow(<App />);
});

test("it renders correctly", () => {
    const shallowRenderer = new ShallowRenderer();
    const snapshotComponent = shallowRenderer.render(<App />);
    expect(snapshotComponent).toMatchSnapshot();
});

test("it contains a Header component", () => {
    expect(component.find(Header)).toHaveLength(1);
});
