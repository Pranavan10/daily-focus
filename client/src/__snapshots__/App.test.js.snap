// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      exact={true}
      path="/"
    >
      <Redirect
        to="/signup"
      />
    </Route>
    <Route
      exact={true}
      path="/signup"
    >
      <LandingPage />
    </Route>
    <Route
      exact={true}
      path="/home"
    >
      <Header />
    </Route>
  </Switch>
</BrowserRouter>
`;
